src/pmdas/statsd/src/pmda-callbacks.c

```diff
@@ -134,6 +134,7 @@ create_pcp_metric(char* key, struct metric* item, pmdaExt* pmda) {
         "STATSD: adding metric %s %s from %s\n", item->meta->pcp_name, pmIDStr(item->meta->pmid), item->name
     );
     item->meta->pcp_metric_index = i;
+    data->pcp_metric_count += 1;
```

Comment:
This incremented total mapped metric count by 1, for each rebuild of PMNS, meaning that it also counted metrics that already had their counterpart pmdaMetric structures in `pmdaRehash(pmda, data->pcp_metrics, data->pcp_metric_count);`'s data->pcp_metrics.